src/services/routeService.ts

```diff
@@ -14,42 +12,37 @@ export interface IMultipleRoutesQueryResult {
 }
 
 export default class RouteService {
-    public static async runFetchRouteQuery(routeId: string): Promise<IRouteResult> {
-        try {
-            const queryResult: ApolloQueryResult<any> = await apolloClient.query(
-                { query: getRouteQuery, variables: { routeId } },
-            );
-            const line = await LineService.getLine(queryResult.data.route.lintunnus);
+    public static async runFetchRouteQuery(routeId: string): Promise<IRouteResult | null> {
+        const queryResult: ApolloQueryResult<any> = await apolloClient.query(
+            { query: getRouteQuery, variables: { routeId } },
+        );
+        const line = await LineService.getLine(queryResult.data.route.lintunnus);
+        if (line !== null) {
             return RouteFactory.createRoute(queryResult.data.route, line);
-        } catch (err) {
-            // tslint:disable-next-line
-            console.error(err);
-            NotificationStore.addNotification({
-                message: 'Reitin haku ei onnistunut.',
-                type: NotificationType.ERROR,
-            });
-            return {
-                nodes: [],
-            };
         }
+        return null;
     }
 
     public static async fetchRoute(routeId: string): Promise<IRoute | undefined> {
-        return (await RouteService.runFetchRouteQuery(routeId)).route;
+        const routeResult = await RouteService.runFetchRouteQuery(routeId);
+        if (routeResult !== null) {
+            return routeResult.route;
+        }
+        return undefined;
```

Comment:
In case of error, the service will return null. The service has already handled the error, so we can silently ignore.